[flake]: test_second_process_blocks_until_first_releases — wall-clock duration assertion is jitter-sensitive on macOS CI

### Summary

``packages/memtomem/tests/test_locking_contention.py::TestAtomicLockContention::test_second_process_blocks_until_first_releases`` (line 132) sometimes fails on macOS CI runners with a sub-50ms shortfall. Most recent occurrence: PR #820 run, where the assertion threshold was 0.25s and the actual wait was 0.197s. The flake has no relationship to the locking implementation — it's a test-shape issue.

```
assert (p2_acquired - p2_requested) >= hold_seconds * 0.5
E   assert (434.218583208 - 434.021797041) >= (0.5 * 0.5)
```

### Root cause

The flaky assertion measures *wall-clock duration* between p2's lock request and p2's lock acquisition, and demands it exceeds ``hold_seconds * 0.5``. On macOS CI runners, ``Process.start()`` jitter can delay p2's request timestamp until well past the midpoint of p1's 0.5s hold window. The lock itself is functioning correctly — p2 *was* blocked — but the measured duration falls below the threshold.

The other assertion in the same test (p2 acquired after p1 released, paraphrasing) is the actually load-bearing check and is jitter-immune.

### Fix sketch

Replace the duration-based assertion with a *phase-ordering* assertion:

- ``p2_requested < p1_released`` — p2 actually tried to acquire while the lock was still held
- ``p2_acquired >= p1_released`` — p2 didn't sneak in before release

This pins the contention semantics without depending on wall-clock duration ratios. The current ``hold_seconds * 0.5`` slack was an attempt to absorb jitter but doesn't bound it correctly on slow spawn paths.

### Scope

Test-only change in ``test_locking_contention.py``. No product impact.

### memtomem version

main

### Operating system

macOS (CI; not reproducible locally most of the time)

### Additional context

- PR #820 macOS run that triggered this report: https://github.com/memtomem/memtomem/actions/runs/25420066180/job/74559959172
- Discussion: the macOS CI failure on #820 was investigated and confirmed unrelated to that PR's Windows-specific changes; this issue tracks the underlying flake.

--- a/packages/memtomem/tests/test_locking_contention.py
+++ b/packages/memtomem/tests/test_locking_contention.py
@@ -124,12 +124,22 @@ class TestAtomicLockContention:
         assert p1.exitcode == 0
         assert p2.exitcode == 0
 
+        # Phase-ordering pin (jitter-immune): p2 made its request while p1
+        # still held the lock, and only acquired once p1 had released. The
+        # earlier ``(p2_acquired - p2_requested) >= hold_seconds * 0.5``
+        # check measured wall-clock wait and flaked on macOS CI when
+        # ``Process.start()`` jitter pushed p2_requested past p1's hold
+        # midpoint — the lock was working, the duration just didn't make
+        # the cutoff (#821).
+        assert p2_requested < p1_released, (
+            f"p2 requested {p2_requested} after p1 released {p1_released} — "
+            "spawn jitter exceeded hold window; the lock can't be proven "
+            "to have blocked p2 in this run"
+        )
         # 50ms scheduler slack: p2 must not have entered before p1 left.
         assert p2_acquired >= p1_released - 0.05, (
             f"p2 acquired {p2_acquired} before p1 released {p1_released}"
         )
-        # And p2 was actually blocked for most of the hold window.
-        assert (p2_acquired - p2_requested) >= hold_seconds * 0.5
 
     def test_uncontended_acquire_is_immediate(self, tmp_path: Path):
         """Negative pin: with no holder, _file_lock acquires without

